viam-cartographer/src/carto_facade/carto_facade.cc

```diff
@@ -689,16 +729,30 @@ void CartoFacade::SaveInternalStateOnInterval() {
 
         std::lock_guard<std::mutex> lk(map_builder_mutex);
         map_builder.SaveMapToFile(true, filename_with_timestamp);
-        if (!started) {
+        if (state != CartoFacadeState::STARTED) {
             LOG(INFO) << "Finished saving final optimized internal state";
             break;
         }
     }
 }
 
-void CartoFacade::Stop() { StopSaveInternalState(); };
+void CartoFacade::Stop() {
+    if (state != CartoFacadeState::STARTED) {
```

Comment:
Can we call Stop when the state is INITIALIZED or IO_INITIALIZED?

Reply:
Nope. Stop() stops the background thread which persists internal state to disk. Calling Stop() in any state other than `STARTED` means that there will be no background thread to stop.